tensorflow_datasets/scripts/refactoring/refactoring_dataset.py

```diff
@@ -0,0 +1,178 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+
+from absl import app
+import os
+import posixpath
+import shutil
+import tensorflow as tf
+from tensorflow_datasets.core import naming
+from typing import Dict
+
+"""A refactoring script to aggregate datasets file from diferent directories
+
+my_dataset/
+    fake_data/
+    __init__.py
+    dataset.py
+    dataset_test.py
+    make_fake_data.py
+    checksum.txt
+
+To run this script
+python3 -m tensorflow_datasets.scripts.refactoring.refactoring_dataset
+
+New datasets will be generated in tensorflow_datasets/refactored_dataset
+
+"""
+
+TFDS_DIR = 'tensorflow_datasets'
+NEW_TFDS_DIR = 'tensorflow_datasets/refactored_dataset'
+URL_CHECKSUM_DIR = 'tensorflow_datasets/url_checksums'
+FAKE_DATA_DIR = 'tensorflow_datasets/testing/test_data/fake_examples'
+FAKE_DATA_SCRIPT_DIR = 'tensorflow_datasets/testing/fake_data_generation'
+
+_INIT_FILE = """\
+\"""{dataset_name} dataset.\"""
+from tensorflow_datasets.{dataset_type}.{dataset_name}.{dataset_name} import {dataset_name_cls}
+"""
+
+DATASET_TYPE = ['audio',
+                'image',
+                'image_classification',
+                'object_detection',
+                'question_answering',
+                'structured',
+                'summarization',
+                'text',
+                'translate',
+                'video']
+
+audio_datasets = ['common_voice', 'crema_d', 'dementiabank', 'fuss', 'groove',
+                  'librispeech', 'libritts', 'ljspeech', 'nsynth', 'savee',
+                  'speech_commands', 'tedlium', 'vctk', 'voxceleb', 'voxforge']
+
+DATASETS_NAMES = {
+  'audio': ['tedlium'],    # will add full audio_datasets list
+  'image': ['abstract_reasoning'],
+  'image_classification': ['binary_alpha_digits']
+}
+
+def create_dirs(dataset_path: str) -> None:
+  """Creates a new `my_dataset` directory"""
+  if not tf.io.gfile.exists(dataset_path):
+    tf.io.gfile.makedirs(os.path.join(dataset_path, 'fake_data'))
+
+
+def copytree(src, dst):
+  """Copy files and folders recursively"""
```

Comment:
What is the difference with `shutil.copytree` ? This should appear in the doc

Reply:
`copytree` function overwrites the contents if present.
there is `dirs_exist_ok=False` parameter in `shutil.copytree` with python3.8. So, to support this functionality with python <3.8, I wrote a custom function to do the same.

Reply:
I'll remove `copytree` function by replacing code in `copy_fake_data_dir` function

```python
def copy_fake_data_dir(src_fake_data_dir, dest_path) -> None:
  """Copy fake data directory"""
  if tf.io.gfile.exists(src_fake_data_dir):
    copytree(src_fake_data_dir, os.path.join(dest_path, 'fake_data'))
```
with

```python
def copy_fake_data_dir(src_fake_data_dir, dest_path) -> None:
  """Copy fake data directory"""
  if tf.io.gfile.exists(src_fake_data_dir):
    if tf.io.gfile.exists(dest_path):
      tf.io.gfile.rmtree(dest_path)
    shutil.copytree(src_fake_data_dir, os.path.join(dest_path, 'fake_data'))
```

Is it good?

Reply:
Done